AMBARI-23348 - service advisor has the option to calculate colocatSe…

dilica opened the pull request:
…rvices with more information from the services list (dili)

## What changes were proposed in this pull request?

Add a new method to the top level service_advisor.py so that each service can either implement colocateService or colocateServiceWithServiceInfo(). The point is to not break existing service advisor implementations in HDP 3.0

## How was this patch tested?

Patch an AMbari 2.7/HDP 3.0 cluster with the changes, add KNox, verify I could add Knox

Please review [Ambari Contributing Guide](https://cwiki.apache.org/confluence/display/AMBARI/How+to+Contribute) before opening a pull request.

Commit:
AMBARI-23348 - service advisor has the option to calculate colocatServices with more information from the services list (dili)

asfgit commented:
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/Ambari-Github-PullRequest-Builder/1353/
Test FAILed.
Test FAILured.

Commit:
AMBARI-23348
service advisor has the option to calculate colocatServices with more information from the services list - fix unit test failures(dili)

dilica commented:
retest this please

asfgit commented:
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/Ambari-Github-PullRequest-Builder/1392/
Test FAILed.
Test FAILured.

asfgit commented:
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/Ambari-Github-PullRequest-Builder/1393/
Test FAILed.
Test FAILured.

dilica commented:
ambari-server Python unit test failure is unrelated to the service advisor change

_FAIL: test_web_server_startup_timeout (TestAmbariServer.TestAmbariServer)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jenkins/jenkins-slave/workspace/Ambari-Github-PullRequest-Builder/ambari-common/src/test/python/mock/mock.py", line 1199, in patched
    return func(*args, **keywargs)
  File "/home/jenkins/jenkins-slave/workspace/Ambari-Github-PullRequest-Builder/ambari-server/src/test/python/TestAmbariServer.py", line 8697, in test_web_server_startup_timeout
    self.assertEquals(50, timeout)
AssertionError: 50 != 90_

dilica commented:
retest this please

jonathan-hurley reviewed (approved):
Looks like this change just adds the contract for this and moves the actual work into a specific service advisor which doesn't exist in trunk anymore - is that correct?

dilica commented:
Hi Jonanthan,

Yes, this is just to provide the necessary method that each service can implement.  HDP services don't need it, current service advisor implementations in HDP 3.0 still work. It's more or less for work on my side, and I don't want to hammer service advisor during my custom service or mpack install.

asfgit commented:
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/Ambari-Github-PullRequest-Builder/1472/
Test PASSed.
